src/workerd/io/worker.c++

```diff
@@ -3243,24 +3241,25 @@ auto Worker::Actor::scheduleAlarm(kj::Date scheduledTime) -> kj::Promise<Schedul
   auto& scheduledAlarm = impl->maybeScheduledAlarm.emplace(
       scheduledTime, kj::newPromiseAndFulfiller<AlarmResult>());
 
-  auto whenCanceled = scheduledAlarm.resultPromise.addBranch().then(
-      [](AlarmResult result) -> ScheduleAlarmResult {
+  // Probably don't need to use kj::coCapture for this but doing so just to be on the
+  // safe side...
+  auto whenCanceled = (kj::coCapture([&scheduledAlarm]() -> kj::Promise<ScheduleAlarmResult> {
     // We've been cancelled, so return that result. Note that we cannot be resolved any other
     // way until we return an AlarmFulfiller below.
-    return result;
-  });
+    co_return co_await scheduledAlarm.resultPromise.addBranch();
```

Comment:
We can `co_await scheduleAlarm.resultPromise` directly now and avoid an allocation (it looks like you take advantage of this elsewhere).